webapp/src/components/NewsItem.vue

```diff
@@ -1,25 +1,32 @@
 <template>
   <div>
-    <h1>{{ title }} {{ (votes) }}</h1>
-    <button v-on:click="incrementVotes">Upvote</button>
-    <button v-on:click="decrementVotes">Downvote</button>
-    <button v-on:click="removeMe">Remove</button>
+    <h1>{{ newsItem.title }} {{ (newsItem.votes) }}</h1>
+    <button @click="incrementVotes">Upvote</button>
+    <button @click="decrementVotes">Downvote</button>
+    <button @click="removeMe">Remove</button>
   </div>
 </template>
 
 <script>
 export default {
   name: "NewsItem",
-  props:["title","votes"],
+  props: {
+    newsItem: {
+      title: {type: String, required: true},
+      votes: {type: String, required: true},
+    }
+  },
   methods: {
     incrementVotes() {
-      this.$emit('upvoteItem', this.title)
+      this.newsItem.votes += 1;
+      this.$emit('upvoteItem', this)
```

Comment:
You're not supposed to `$emit` the entire component, but some object, e.g. a reactive state.